components/admin/lessons/AdminLessonSideNav.tsx

```diff
@@ -0,0 +1,129 @@
+import React from 'react'
+import { ApolloError } from '@apollo/client'
+import { AlertFillIcon, ChevronRightIcon } from '@primer/octicons-react'
+import { isNumber } from 'lodash'
+import { Spinner } from 'react-bootstrap'
+import { get } from 'lodash'
+import styles from '../../../scss/adminLessonSideNav.module.scss'
+
+type Item = {
+  id: number
+  name: string
+  content: string
+  lesson: { id?: number }
+}
+
+type Items = Item[]
+
+type Props = {
+  items: Items | undefined
+  title: string
+  lessonId: number
+  onAddItem: (m: Item | undefined) => void
+  loading?: boolean
+  error?: ApolloError
+  active?: number
+  onClick: (item: Item) => void
+}
+
+const AdminLessonSideNav = ({
+  items,
+  onAddItem,
+  loading,
+  error,
+  active,
+  title,
+  lessonId,
+  onClick
+}: Props) => {
+  const itemsList =
+    items &&
+    items.reduce((acc: JSX.Element[], item) => {
+      if (item.lesson.id === lessonId) {
+        const isActive = active === item.id
+        const className =
+          styles[
+            isActive
+              ? 'container__modulesList__module--active'
+              : 'container__modulesList__module'
+          ]
+        const withChevron = !isActive && <ChevronRightIcon size={16} />
+
+        acc.push(
+          <li className={className} key={item.id} onClick={() => onClick(item)}>
+            <span>{item.name}</span>
+            {withChevron}
+          </li>
+        )
+      }
+
+      return acc
+    }, [])
+
+  const renderItems = () => {
+    if (error) {
+      return (
+        <div className={styles.container__error}>
+          <AlertFillIcon />
+          <span className={styles.container__heading}>
+            Failed to get the modules
```

Comment:
this component is not sending any requests, so it is confusing for it to have a state that says `Failed to get modules` because its not even getting the modules.